### ISSUE TITLE
You can't access non-default Postgresql databases

### ISSUE THREAD

**Issue by GiantCrocodile** [dbeaver/dbeaver#2131]
I get this error whenever I try to interact with my non-default created database:

```
org.jkiss.dbeaver.DBException: Can't access non-default database
	at org.jkiss.dbeaver.ext.postgresql.model.PostgreDatabase.getSchemas(PostgreDatabase.java:256)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.extractPropertyValue(DBNDatabaseNode.java:793)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadTreeItems(DBNDatabaseNode.java:479)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.loadChildren(DBNDatabaseNode.java:403)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:190)
	at org.jkiss.dbeaver.model.navigator.DBNDatabaseNode.getChildren(DBNDatabaseNode.java:1)
	at org.jkiss.dbeaver.ui.navigator.NavigatorUtils.getNodeChildrenFiltered(NavigatorUtils.java:563)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:49)
	at org.jkiss.dbeaver.ui.navigator.database.load.TreeLoadService.evaluate(TreeLoadService.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:86)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:71)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:95)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

```

Is this a fault of me or of DBeaver? I just created it with `CREATE DATABASE "<name>" OWNER postgres`

**Comment by GiantCrocodile** [comment]
I just found out that I have to set current database active manually: context menu of database -> Set Active. I would expect this to happen automatically if I try to navigate in the tabs of this database or when I do double-clicks on the database name in navigator.

**Comment by kevinhaas** [comment]
I also expected that behavior. Is there something we're missing here? Every tool I've ever used has set the currently selected DB to 'active' automatically.

**Comment by serge-rider** [comment]
PostgreSQL is special. Different databases require different physical connections. You can't just switch active database in current connection. Changing active database means disconnect+connect, loose of all current transactions, different security, etc.

Most of tools doesn't allow PG database switch at all. In DBeaver there is a special workaround. Perhaps it should be additionally improved..

**Comment by kevinhaas** [comment]
Thanks for the quick response.

I'm coming from pgAdmin and it works how I described. It's not a huge deal, and the program functions much better than pgAdmin so far. Just something that would be a nice QoL improvement.

**Comment by serge-rider** [comment]
Yep, pgAdmin is PG-specific tool. It supports most of PG features. 
DBeaver pays for universality :(

**Comment by antimirov** [comment]
How do other DB tools handle this? Do they really make 10-20 connection when you work with multiple databases on the same server? I remember on the server side I always saw only one network connection.

**Comment by serge-rider** [comment]
Yes, any PostgreSQL tool (including command line psql) creates separate connection (at least one) for different PG databases. This is how PostgreSQL works.

**Comment by tati-kru** [comment]
There isn't any activity about this ticket very long time. It will be closed. However, if you think that it is actual just let us know.

**Comment by GiantCrocodile** [comment]
Is it really a good idea to close tickets without activity just because they are old? This is still in fact a requested feature and I highly doubt it is wise to
a) to snub someone by closing a discussion by just closing the ticket because it is old (this is rude to the persons who take their time to report and discuss the issue) and
b) discuss the same topic all over again, just because there is some old issue which is out of focus of the devs, simply because the issue is unresolved closed.

@tati-kru @serge-rider

**Comment by serge-rider** [comment]
I'm pretty sure we need a new ticket. It is completely not clear what to do in this one.
As in PG different DBs are actually different connections we can't  make this UI similar to other database systems.
There was no any UI solution proposed. Or a workaround. It is a kind of bug report but it is not a bug. So this ticket may lay here forever and depressing us (as we have too many open tickets).

Let's create a new ticket with some proposed solution, or at least with some discussion which will let us to find a solution - this will make everybody happy.

And I'm sure that @tati-kru didn't mean any offence here, sorry if it was a bit frustrating.

**Comment by AndrewKhitrin** [comment]
Agree. New ticket its a good idea if we need discussion. BTW   current functionality for PG database is the only way to implement "change active" scenario

**Comment by GiantCrocodile** [comment]
Nevermind then, thanks for clear up and open communication on feedback!

**Comment by serge-rider** [comment]
Resolved in #2892

**Comment by khairul-abdi** [comment]
I get this error whenever I try to make data type Timestamp created database:

Navigator node 'database://postgres-jdbc-16fa8e8bc40-47a9661345f0723/codegig/org.jkiss.dbeaver.ext.postgresql.model.PostgreSchema/public/org.jkiss.dbeaver.ext.postgresql.model.PostgreTable/gigs/org.jkiss.dbeaver.ext.postgresql.model.PostgreTableColumn/updatedAt' not found

More Details: 
org.jkiss.dbeaver.DBException: Navigator node 'database://postgres-jdbc-16fa8e8bc40-47a9661345f0723/codegig/org.jkiss.dbeaver.ext.postgresql.model.PostgreSchema/public/org.jkiss.dbeaver.ext.postgresql.model.PostgreTable/gigs/org.jkiss.dbeaver.ext.postgresql.model.PostgreTableColumn/createdAt' not found
	at org.jkiss.dbeaver.ui.editors.DatabaseLazyEditorInput.initializeRealInput(DatabaseLazyEditorInput.java:270)
	at org.jkiss.dbeaver.ui.editors.ProgressEditorPart$InitNodeService.evaluate(ProgressEditorPart.java:134)
	at org.jkiss.dbeaver.ui.editors.ProgressEditorPart$InitNodeService.evaluate(ProgressEditorPart.java:1)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:86)
	at org.jkiss.dbeaver.ui.LoadingJob.run(LoadingJob.java:71)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:103)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
